src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/services/NomisApiService.kt

```diff
@@ -78,12 +79,20 @@ class NomisApiService(
 
   // ////////// VISITS //////////////
 
-  suspend fun createVisit(request: CreateVisitDto): CreateVisitResponseDto =
+  suspend fun createVisit(request: CreateVisitDto): Result<CreateVisitResponseDto> =
     webClient.post()
       .uri("/prisoners/{offenderNo}/visits", request.offenderNo)
       .bodyValue(request)
       .retrieve()
-      .awaitBody()
+      .bodyToMono(CreateVisitResponseDto::class.java)
+      .map { Result.success(it) }
+      .onErrorResume(WebClientResponseException.Conflict::class.java) {
+        val errorResponse = it.getResponseBodyAs(ErrorResponse::class.java) as ErrorResponse
+        Mono.just(Result.failure(CreateVisitDuplicateResponse(nomisVisitId = errorResponse.moreInfo!!)))
+      }
+      .awaitSingle()
+
```

Comment:
409 will return error response with moreInfo containing the existing NOMIS visit i